xbmc/games/controllers/types/ControllerNode.cpp

```diff
@@ -79,9 +82,14 @@ void CControllerNode::GetControllers(ControllerVector& controllers) const
   m_hub->GetControllers(controllers);
 }
 
-void CControllerNode::SetAddress(std::string address)
+void CControllerNode::SetPortAddress(std::string portAddress)
```

Comment:
just want to remark i'd prefer the pattern `const std::string& portAddress + no move` but this is your playground

Reply:
Valid preference. Why do you prefer const reference?

I like move semantics because it pushes the memory copy outside the function, and control over memory management is delegated to the caller. As a result you have lighter-weight functions with less side effects. The caller knows best about its memory, so it can take advantage of move semantics too if it wants and avoid a copy.

One of our big RetroPlayer projects was moving memory management outside the function and delegating it to the caller. Needed for zero-copy, obviously a concern when you have thousands of pixels, less so when it's just a string. The inconsequentialness of a string copy is why I think individual preference is ok instead of a hard-cut rule.

Reply:
well, it's just a vague rule of thumb that sunk into me: pass primitive types by value and any other object by (const) reference. but you're right, things like `obj.SetSomething(std::move(hugeObjThatHasAMoveCtor));` or even passing in a temporary is only possible with your function signature. (if i got it right now)... learned something new. great. thanks :)

Reply:
It's a little confusing, you could do `void SetSomething(huge&& h) { … }`, but that would _force_ the caller to use move semantics, whereas with the approach in the PR, the caller has a choice. `SetSomething(huge h)` effectively calls the `huge::huge(huge&& h)` constructor when the function is called via `SetSomething(std::move(h))`.